src/live_restore/live_restore_fs.c

```diff
@@ -1248,6 +1184,131 @@ __live_restore_handle_verify_hole_list(WT_SESSION_IMPL *session, WTI_LIVE_RESTOR
     return (ret);
 }
 
+/*
+ * __wt_live_restore_fh_import_extents_from_string --
+ *     Reconstruct the extent list in memory from a string representation. If the string is NULL
+ *     mark the destination as complete. On error free any allocated extents.
+ */
+int
+__wt_live_restore_fh_import_extents_from_string(
+  WT_SESSION_IMPL *session, WT_FILE_HANDLE *fh, const char *extent_str)
+{
+    WT_DECL_RET;
+    WTI_LIVE_RESTORE_FILE_HANDLE *lr_fh = (WTI_LIVE_RESTORE_FILE_HANDLE *)fh;
+
+    if (!F_ISSET(S2C(session), WT_CONN_LIVE_RESTORE_FS))
+        return (0);
+
+    /*
+     * This function can be called for file handles that already have an in memory extent list. For
+     * this to happen the destination file was created for the first time and a single file size
+     * hole was initialized.
+     *
+     * There is a tricky scenario here:
+     *   - Open a file that exists in the source, a.wt.
+     *   - Create a new file in the destination to begin migrating the file to.
+     *   - Crash.
+     *   - Open the file a.wt again, we will see an a.wt in the destination and not create the
+     *   necessary file length hole. We will also get an empty extent list string indicating a.wt is
+     *   complete.
+     */
+    bool extent_string_empty = extent_str == NULL || strlen(extent_str) == 0;
+
+    if (lr_fh->destination.hole_list_head != NULL) {
+        WT_ASSERT_ALWAYS(
+          session, extent_string_empty, "Extent list not empty while trying to import");
+        return (0);
+    }
+
+    if (extent_string_empty)
+        lr_fh->destination.complete = true;
+    else {
+        __wt_readlock(session, &lr_fh->ext_lock);
+        __wt_verbose_debug3(session, WT_VERB_LIVE_RESTORE, "%s metadata extent list string: %s",
+          fh->name, extent_str);
+        /* The extents are separated by ;. And have the shape %d-%u. */
+        wt_off_t off = 0, next_off;
+        size_t len;
+        WTI_LIVE_RESTORE_HOLE_NODE **current = &lr_fh->destination.hole_list_head;
+        const char *str_ptr = extent_str;
+        char *next;
+        while (true) {
+            if (*str_ptr > '9' || *str_ptr < '0')
+                WT_ERR_MSG(session, EINVAL, "Invalid extent string found");
+
+            next_off = (wt_off_t)strtoll(str_ptr, &next, 10);
+            str_ptr = next;
+            if (*str_ptr == '\0')
+                WT_ERR_MSG(session, EINVAL, "Invalid extent string found");
+
+            off += next_off;
```

Comment:
The unit test has a comment that mentions 
> "Extents are additive to compress the string size. I.e. the offset of extent 1 is the offset of extent 0 + the offset of extent 1." 

Can we have a similar explanation here?